cmd/aries-agent-lambda/aries.go

```diff
@@ -0,0 +1,387 @@
+	inboundWsHost := fmt.Sprintf("ws@%s", wsHost)
+	log.Printf("Inbound Ws Host: %s", inboundWsHost)
+	inboundHostHttpExternalUrl := fmt.Sprintf("http@%s://%s/%s", httpProto, httpHost, basePath)
+	log.Printf("Inbound Http Host External URL: %s", inboundHostHttpExternalUrl)
+	inboundHostWsExternalUrl := fmt.Sprintf("ws@%s://%s/%s", wsProto, wsHost, basePath)
+	log.Printf("Inbound Ws Host External URL: %s", inboundHostWsExternalUrl)
+	// TODO: Convert defaultWebhookUrl to Environment Variables
+	webhookUrls := []string{defaultWebhookUrl, fmt.Sprintf("%s://%s%s", httpProto, internalHttpHost, webhookPath)}
+	log.Printf("Webhook URLs: %s", webhookUrls)
+
+	parameters := &agentParameters{
+		// server:               server,
+		host: httpHost,
+		// token:                token,
+		inboundHostInternals: []string{inboundWsHost},
+		inboundHostExternals: []string{inboundHostWsExternalUrl},
+		dbParam:              dbParam,
+		// defaultLabel:         defaultLabel,
+		webhookURLs: webhookUrls,
+		// httpResolvers:        httpResolvers,
+		outboundTransports: []string{"ws"},
+		autoAccept: true,
+		transportReturnRoute: "all",
+		// contextProviderURLs:  contextProviderURLs,
+		// tlsCertFile:          tlsCertFile,
+		// tlsKeyFile:           tlsKeyFile,
+		// autoExecuteRFC0593:   autoExecuteRFC0593,
+		// keyType:              keyType,
+		// keyAgreementType:     keyAgreementType,
+		// mediaTypeProfiles:    mediaTypeProfiles,
+	}
+
+	parameters.msgHandler = msghandler.NewRegistrar()
+
+	actx, err := createAriesAgent(parameters)
+	if err != nil {
+		return nil, err
+	}
+
+	handlers, err := controller.GetRESTHandlers(actx, controller.WithWebhookURLs(parameters.webhookURLs...),
+		controller.WithDefaultLabel(parameters.defaultLabel), controller.WithAutoAccept(parameters.autoAccept),
+		controller.WithMessageHandler(parameters.msgHandler),
+		controller.WithAutoExecuteRFC0593(parameters.autoExecuteRFC0593))
+	if err != nil {
+		return nil, err
+	}
+
+	if parameters.token != "" {
+		router.Use(authorizationMiddleware(parameters.token))
+	}
+
+	for _, handler := range handlers {
+		path := fmt.Sprintf("/%s%s", basePath, handler.Path())
+		//log.Printf("Registering Path: %s %s", handler.Method(), handler.Path())
+		router.HandleFunc(path, handler.Handle()).Methods(handler.Method())
+	}
+
+	router.HandleFunc(webhookPath, webhookHandler).Methods(http.MethodPost)
+	router.HandleFunc("/ws", wsinternal.NewInboundHandler()).Methods(http.MethodPut, http.MethodPost, http.MethodDelete)
+	router.HandleFunc(fmt.Sprintf("/%s", basePath), httpinternal.NewInboundHandler()).Methods(http.MethodPost)
+
+	return router, nil
+}
+
+func createAriesAgent(parameters *agentParameters) (*ariescontext.Provider, error) {
+	var opts []aries.Option
+
+	storePro := dynamodb.NewProvider() // TODO Other Storage Providers
+
+	opts = append(opts, aries.WithStoreProvider(storePro))
+	opts = append(opts, aries.WithProtocolStateStoreProvider(storePro))
```

Comment:
We're thinking of pushing this to Redis, or other type of temporary storage. For now, DynamoDB does make it work nicely.